### PR TITLE
Add normalizeStopTimes flag to RDBMS snapshots preserve stop_sequence vals

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

re #283

### CODE REVIEW

File: src/main/java/com/conveyal/gtfs/loader/JdbcGtfsSnapshotter.java [hunk 1]
```diff
@@ -49,10 +56,13 @@
     /**
      * @param feedId namespace (schema) to snapshot. If null, a blank snapshot will be created.
      * @param dataSource the JDBC data source with database connection details
+     * @param normalizeStopTimes whether to keep stop sequence values intact or normalize to be zero-based and
+     *                           incrementing
      */
-    public JdbcGtfsSnapshotter(String feedId, DataSource dataSource) {
+    public JdbcGtfsSnapshotter(String feedId, DataSource dataSource, boolean normalizeStopTimes) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/loader/JdbcGtfsSnapshotter.java:62]: Should this new parameter be renamed `normalizeStopTimesSequence` or something similar? that's what this parameter utlimately controls in the normalizeAndCloneStopTimes method.

File: src/main/java/com/conveyal/gtfs/GTFS.java [hunk 2]
```diff
@@ -81,16 +81,24 @@ public static FeedLoadResult load (String filePath, DataSource dataSource) {
...
-     * @param dataSource    JDBC connection to existing database
-     * @return              FIXME should this be a separate SnapshotResult object?
+     * @param feedId                feed ID (schema namespace) to copy from
+     * @param dataSource            JDBC connection to existing database
+     * @param normalizeStopTimes    whether to normalize stop sequence values on snapshot
+     * @return the result of the snapshot
      */
-    public static SnapshotResult makeSnapshot (String feedId, DataSource dataSource) {
-        JdbcGtfsSnapshotter snapshotter = new JdbcGtfsSnapshotter(feedId, dataSource);
+    public static SnapshotResult makeSnapshot (String feedId, DataSource dataSource, boolean normalizeStopTimes) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/GTFS.java:89]: Should this new parameter be renamed normalizeStopTimesSequence or something similar? that's what this parameter utlimately controls in the normalizeAndCloneStopTimes method.

File: src/main/java/com/conveyal/gtfs/loader/Table.java [hunk 3]
```diff
@@ -834,16 +834,19 @@ public void createIndexes(Connection connection, String namespace) throws SQLExc
     /**
      * Creates a SQL table from the table to clone. This uses the SQL syntax "create table x as y" not only copies the
      * table structure, but also the data from the original table. Creating table indexes is not handled by this method.
+     *
+     * Note: the stop_times table is a special case that will optionally normalize the stop_sequence values to be
+     * zero-based and incrementing.
      */
-    public boolean createSqlTableFrom(Connection connection, String tableToClone) {
+    public boolean createSqlTableFrom(Connection connection, String tableToClone, boolean normalizeStopTimes) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/loader/Table.java:841]: Add JavaDoc for the new parameter.

File: src/test/java/com/conveyal/gtfs/GTFSTest.java [hunk 4]
```diff
@@ -428,21 +426,12 @@ private boolean runIntegrationTestOnZipFile(
...
-            TestUtils.dropDB(testDBName);
-            return false;
-        } catch (AssertionError e) {
-            TestUtils.dropDB(testDBName);
-            throw e;
-        }
+        boolean snapshotIsOk = assertThatSnapshotIsSuccessful(namespace, dataSource, testDBName, persistenceExpectations, false);
+        if (!snapshotIsOk) return false;
+        // Also, verify that if we're normalizing stop_times#stop_sequence
+        PersistenceExpectation[] expectationsWithNormalizedStopTimes =  updatePersistenceExpectationsWithNormalizedStopTimes(persistenceExpectations);
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/gtfs/GTFSTest.java:432]: Here we can definitely say `expectationsWithNormalizedStopTimesSequence`

File: src/test/java/com/conveyal/gtfs/GTFSTest.java [hunk 5]
```diff
@@ -428,21 +426,12 @@ private boolean runIntegrationTestOnZipFile(
...
-            assertThatExportedGtfsMeetsExpectations(tempFile, persistenceExpectations, true);
-        } catch (IOException e) {
-            e.printStackTrace();
-            TestUtils.dropDB(testDBName);
-            return false;
-        } catch (AssertionError e) {
-            TestUtils.dropDB(testDBName);
-            throw e;
-        }
+        boolean snapshotIsOk = assertThatSnapshotIsSuccessful(namespace, dataSource, testDBName, persistenceExpectations, false);
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/gtfs/GTFSTest.java:429]: For this test and the one just below, I kinda want to write `if (!assert...()) return false`.

File: src/test/java/com/conveyal/gtfs/GTFSTest.java [hunk 6]
```diff
@@ -798,127 +812,147 @@ private void assertThatPersistenceExpectationRecordWasFound(
...
+                new RecordExpectation("bikes_allowed", 0),
+                new RecordExpectation("wheelchair_accessible", 0)
+            }
+        )
     };
+
+    /**
+     * Update persistence expectations to expect normalized stop_sequence values.
+     */
+    private PersistenceExpectation[] updatePersistenceExpectationsWithNormalizedStopTimes(PersistenceExpectation[] inputExpectations) {
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/gtfs/GTFSTest.java:940]: `updatePersistenceExpectationsWithNormalizedStopTimesSequence`

File: src/test/java/com/conveyal/gtfs/GTFSTest.java [hunk 7]
```diff
@@ -798,127 +812,147 @@ private void assertThatPersistenceExpectationRecordWasFound(
...
+        PersistenceExpectation[] persistenceExpectations = new PersistenceExpectation[inputExpectations.length];
+        // Add all of the table expectations.
+        for (int i = 0; i < inputExpectations.length; i++) {
+            // Collect record expectations.
+            RecordExpectation[] newRecordExpectations = new RecordExpectation[inputExpectations[i].recordExpectations.length];
+            for (int j = 0; j < inputExpectations[i].recordExpectations.length; j++) {
+                newRecordExpectations[j] = inputExpectations[i].recordExpectations[j].clone();
```
Comment (binh-dam-ibigroup) [inline src/test/java/com/conveyal/gtfs/GTFSTest.java:947]: Create variables for `inputExpectations[i]`, `newRecordExpections[j]`.

Comment (binh-dam-ibigroup) [reply src/test/java/com/conveyal/gtfs/GTFSTest.java:947]: Or some of these could become a map, your choice.

File: src/main/java/com/conveyal/gtfs/loader/JdbcGtfsSnapshotter.java (hunk 1 above)
Comment (landonreed) [reply src/main/java/com/conveyal/gtfs/loader/JdbcGtfsSnapshotter.java:62]: I think `normalizeStopTimes` is descriptive enough with good javadoc. The language 'normalize stop times' is used a number of times here to mean a single, well-documented thing.

**Pull request merged**